src/NLog/Common/InternalLogger.cs

```diff
@@ -198,7 +203,7 @@ public static void Log(Exception ex, LogLevel level, [Localizable(false)] string
         /// <param name="args">optional args for <paramref name="message"/></param>
         private static void Write([CanBeNull]Exception ex, LogLevel level, string message, [CanBeNull]object[] args)
         {
-            if (ex != null && ex.MustBeRethrownImmediately())
+            if (IsSeriousException(ex))
```

Comment:
A little. Code should be readable (like a book) so properly called method tald us  exactly what we check. That's why I extractd method. But this is my proposition so you decide.